modules/services/educator_aaa.nix

```diff
@@ -0,0 +1,80 @@
+{ config, lib, pkgs, ... }: with lib;
+
+let
+  cfg = config.services.educator_aaa;
+  keys = config.dscp.keys;
+  package = pkgs.educator_aaa;
+  stateDirName = "educator_aaa";
+  stateDir = "/var/lib/${stateDirName}";
+in
+
+{
+  options.services.educator_aaa = {
+    enable = mkEnableOption "Educator AAA";
+
+    virtualHost = mkOption {
+      default = "educator_aaa";
+      type = types.nullOr types.string;
+    };
+
+    port = mkOption {
+      type = types.int;
+    };
+  };
+
+  config = mkIf cfg.enable {
+    users.users.educator_aaa = {
+      home = "/var/lib/${stateDir}";
+      createHome = true;
+      isSystemUser = true;
+    };
+
+    services.postgresql = {
+      enable = true;
+      package = pkgs.postgresql100;
+      initialScript = pkgs.writeText "educator_aaa.psql" ''
+        CREATE ROLE "educator_aaa" LOGIN PASSWORD 'educator_aaa';
+        CREATE DATABASE "educator_aaa" OWNER 'educator_aaa';
```

Comment:
I think this conflicts with any other services trying to create databases for themselves, but there doesn't seem to be an option for this that doesn't conflict. Please advice.

Reply:
Yea, `initialScript` is only ever run once for a postgres deployment. But I think it's type `lines`, so it'll merge things before running.

However, `intialScript` is less than reliable for setting up a known state of the database, and I don't like it at all.